tests/test_retention_policies.py

```diff
@@ -249,21 +290,259 @@ def test_first_backup(self, server):
         )
         assert isinstance(rp, RedundancyRetentionPolicy)
 
-        # Build a BackupInfo object with status to DONE
-        backup_info = build_test_backup_info(
-            server=server, backup_id="test1", end_time=datetime.now(tzlocal())
-        )
-
         # instruct the get_available_backups method to return a map with
         # our mock as result and minimum_redundancy = 1
-        server.get_available_backups.return_value = {"test_backup": backup_info}
         server.config.minimum_redundancy = 1
 
         # execute retention policy report
         report = rp.first_backup()
 
         assert report == "test_backup"
 
+    @pytest.mark.parametrize(
+        "retention_policy", ("RECOVERY WINDOW OF 4 WEEKS", "REDUNDANCY 2")
+    )
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        ("retention_policy", "retention_status"),
        itertools.product(
            ("RECOVERY WINDOW OF 4 WEEKS", "REDUNDANCY 2"),
            (
                BackupInfo.OBSOLETE,
                BackupInfo.VALID,
                BackupInfo.POTENTIALLY_OBSOLETE,
                BackupInfo.KEEP_FULL,
                BackupInfo.KEEP_STANDALONE,
                BackupInfo.NONE,
            ),
        ),
    )
```

Instead of using a fixed `RETENSION_STATUS`, let's play with actual valid values for `_propagate_retention_status_to_children` method and check how the method behaves when it receives them.

Here I'm only providing how to parametrize the test. After you apply this suggestion you will need to change the unit test itself to use this new parameter.